packages/babel-plugin/src/utils/css-builders.tsx

```diff
@@ -121,22 +175,25 @@ const extractObjectExpression = (node: t.ObjectExpression, meta: Metadata): CSSO
  * @param state Babel state - should house options and meta data used during the transformation.
  */
 const extractTemplateLiteral = (node: t.TemplateLiteral, meta: Metadata): CSSOutput => {
-  let variables: CSSOutput['variables'] = [];
+  const variables: CSSOutput['variables'] = [];
+  const css: CSSOutput['css'] = [];
+
   // quasis are the string pieces of the template literal - the parts around the interpolations.
-  const css = node.quasis.reduce((css, q, index) => {
+  const literalResult = node.quasis.reduce<string>((acc, q, index): string => {
     const nodeExpression = node.expressions[index];
     const { value: interpolation, meta: updatedMeta } = evaluateExpression(nodeExpression, meta);
 
     if (t.isStringLiteral(interpolation) || t.isNumericLiteral(interpolation)) {
       // Simple case - we can immediately inline the value.
-      return css + q.value.raw + interpolation.value;
+      return acc + q.value.raw + interpolation.value;
     }
 
     if (t.isObjectExpression(interpolation)) {
       // We found an object like: css`${{ red: 'blue' }}`.
-      const result = extractObjectExpression(interpolation, updatedMeta);
-      variables = variables.concat(result.variables);
-      return css + result.css;
+      const result = buildCss(interpolation, updatedMeta);
+      variables.push(...result.variables);
+      css.push(...result.css);
```

Comment:
nit: We have lots of `variables.push(...result.variables);` and `css.push(...result.css);` now. Can we extract them into a function which gives us final result? If its getting complex, we can skip this.